### DISCUSSION TITLE
HiFiGan training from scratch - computational requirements

### DISCUSSION THREAD

**Discussion by oytunturk** [NVIDIA-NeMo/NeMo#3601] [Q&A] [answered]
Hi,

Can you provide some guidance on GPU memory and training time requirements for training HiFiGan neural vocoder from scratch for LJSpeech data?

Thanks!

**Comment by titu1994** [comment]
@blisc could you provide some info

**Comment by qGentry** [comment, marked as answer]
Default setup (16 batch size, 8192 segments) requires ~7Gb GPU memory for fp32 training.
As for training speed, single rtx 3090 can do like 0.6s per iteration, 1M iterations is probably enough for LJSpeech -> ~7 GPU-days, distributed training with 2x3090 (batch size per GPU = 8) gave me 0.45s per iteration -> ~5,25 Days   

I also tried fp16 training, which decreases memory usage by factor of 2 and increases training speed by factor of 1.5 on single GPU (3090), but some of my experiments ended with NaN loss (while some ended with same quality as fp32 experiment)

**Reply by oytunturk** [reply]
Thanks, this is very informative! Do you have any suggestions for a good strategy to train a hifigan model for a male speaker, i.e. fine-tuning an existing female model vs training from scratch? How about using multiple speakers? I'm guessing the model should be modified to support some kind of speaker embedding or speaker ID input to work consistently across different speakers.

**Reply by blisc** [reply]
Hifigan works fine on multiple speakers without speaker ids. You should be able to train a hifigan model on the entirety of HiFiTTS without needing speaker embeddings.

Speaker embeddings might help but I don't know of any experiments adding them to hifigan.

You can try finetuning a new speaker from the baseline LJSpeech checkpoint. I believe it should be fine.

**Reply by oytunturk** [reply]
Thanks, I'll give it a try then.

**Reply by qGentry** [reply]
It depends on amount of data that you have. If you have like 20h+, it's probably won't make any difference if you try to finetune or train from scratch, but if you have less than 10h finetuning on LJSpeech pretrained model most likely end with better sound quality.
Also you can try finetuning universal_v1 model from jik876 repo, but you'll have to write checkpoint converter to use it in NeMo (most keys match, but there are some difference in naming of res modules)

**Reply by oytunturk** [reply]
Makes sense, thanks for the feedback!

**Reply by godspirit00** [reply]
> It depends on amount of data that you have. If you have like 20h+, it's probably won't make any difference if you try to finetune or train from scratch, but if you have less than 10h finetuning on LJSpeech pretrained model most likely end with better sound quality. Also you can try finetuning universal_v1 model from jik876 repo, but you'll have to write checkpoint converter to use it in NeMo (most keys match, but there are some difference in naming of res modules)

@qGentry Could you please explain a little bit more about how to do the conversion? Thank you!

**Comment by Arjunprasaath** [comment]
Guys Im training Tacotron2 and would like to plot attention and losses graph, any idea how to do it?
